packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -629,11 +630,16 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
   // This should be called to wrap any changes to route.isCurrent, route.canPop,
   // and route.offstage.
   void _routeSetState(VoidCallback fn) {
+    if (widget.route.isCurrent) {
+      widget.route.navigator.focusScopeNode.setFirstFocus(focusScopeNode);
+    }
     setState(fn);
   }
 
   @override
   Widget build(BuildContext context) {
+    final bool isReversing = widget.route.animation?.status == AnimationStatus.reverse;
+    focusScopeNode.canRequestFocus = !isReversing;
```

Comment:
What if we're in the middle of an iOS-back-swipe-route-dismiss gesture, i.e. what if route.navigator.userGestureInProgress is true?  Seems like we wouldn't want to take the focus then either?